### PR TITLE
Add ability to define blocks with a json array

### PR DESCRIPTION
**Pull request by rachel-fenichel**

### CODE REVIEW

File: core/blockly.js [hunk 1]
```diff
@@ -399,6 +399,33 @@ Blockly.prompt = function(message, defaultValue, callback) {
   callback(window.prompt(message, defaultValue));
 };
 
+/**
+ * Helper function for defining a block from json.  The resulting function has
```
Comment (NeilFraser) [inline core/blockly.js]: JSON

File: core/blockly.js [hunk 2]
```diff
@@ -399,6 +399,33 @@ Blockly.prompt = function(message, defaultValue, callback) {
...
+ * the correct value of jsonDef at the point in code where jsonInit is called.
+ * @param {Object} jsonDef The JSON definition of a block.
```
Comment (NeilFraser) [inline core/blockly.js]: !Object

File: core/blockly.js [hunk 3]
```diff
@@ -399,6 +399,33 @@ Blockly.prompt = function(message, defaultValue, callback) {
...
+Blockly.jsonInitFactory_ = function(jsonDef) {
+  return function() {
+    this.jsonInit(jsonDef);
+  };
+};
+
+/**
+ * Define blocks from an array of JSON block definitions, as might be generated
+ * by the Blockly Developer Tools.
+ * @param {!Array<Object>} jsonArray An array of JSON block definitions.
```
Comment (NeilFraser) [inline core/blockly.js]: The style guide recently make dots optional in array descriptors, but Blockly is currently consistent in using dots.  I have no objection if we remove all dots, but let's be consistent.  Also, !Object.

File: core/blockly.js [hunk 4]
```diff
@@ -399,6 +399,33 @@ Blockly.prompt = function(message, defaultValue, callback) {
...
+ */
+Blockly.defineBlocksWithJsonArray = function(jsonArray) {
+  for (var index = 0; index < jsonArray.length; index++) {
+    var elem = jsonArray[index];
```
Comment (NeilFraser) [inline core/blockly.js]: Why not just:
for (var i = 0, elem; elem = jsonArray[i]; i++) {

File: core/blockly.js [hunk 5]
```diff
@@ -399,6 +399,33 @@ Blockly.prompt = function(message, defaultValue, callback) {
...
 };
 
+/**
+ * Helper function for defining a block from json.  The resulting function has
+ * the correct value of jsonDef at the point in code where jsonInit is called.
+ * @param {Object} jsonDef The JSON definition of a block.
+ * @return {function} A function that calls jsonInit with the correct value
+ *     of jsonDef.
+ */
+Blockly.jsonInitFactory_ = function(jsonDef) {
```
Comment (NeilFraser) [inline core/blockly.js]: This function is @private.

**Comment by rachel-fenichel** [comment]
It's pretty easy to put an object into an array if you only have one, and the main use case for this is bulk importing from the block factory, which gives you an array anyway.  Leaving it as is.

**Pull request merged**